File: Server/models/Order.js
const { v4: uuidv4 } = require('uuid');

class Order {
    constructor(userId, date, restaurantId, qrCode, tip, foodItems) {
        this.userId = userId;
        this.date = date;
        this.restaurantId = restaurantId;
        this.qrCode = qrCode;
        this.tip = tip;
        this.foodItems = foodItems;
        this.id = uuidv4();
    }
}

module.exports = Order;

File: Dyne/Dyne/Model/Order.swift
//
//  Order.swift
//  Dyne
//
//  Created by Marcelo Costa on 10/28/20.
//  Copyright © 2020 Malek Amiri. All rights reserved.
//

import Foundation

struct SimpleFoodItem {
    var foodItemId: String
    var count: Int
}

class Order {
    
    var foodItems: [SimpleFoodItem]
    var date: Date
    var userID: String
    var restID: String
    var location: String
    var tip: Double
    
    init(foodItems: [SimpleFoodItem], date: Date, userID: String, restID: String, location: String, tip: Double) {
        self.foodItems = foodItems;
        self.date = date;
        self.userID = userID;
        self.restID = restID;
        self.location = location;
        self.tip = tip;
    }
}
